v3.1

timoniq opened the pull request:
* [x] Ваш код документирован.
* [x] Для вашего кода есть тесты.
* [x] Для вашего кода есть примеры использования в examples.

Lisovq committed:
User implementation (#327)

* Add User implementation

* Hot fix (message_ids param in method messages.getById contains full user-event, not message_id)

* general refactor

* rules now generics and stored in `base`
* router is now common for user and bot
* resolve mypy issues
* resolve flake issues

Co-authored-by: timoniq <tesseradecades@mail.ru>

timoniq committed:
fix event id unwrap

timoniq committed:
fix task_each_event defaults

timoniq committed:
update readme

timoniq committed:
update workflows

timoniq committed:
fix typings

dhvcc committed:
Redo middleware (#343)

* Cherry-pick and squash previous changes

* Updated middleware/view docs, updated tests

Updated BaseMiddleware and ABCView
Fixed Makefile help target

* Moved args from pre/post to instance, updated views sugar

Added "==" sugar for FromFuncHandler
Updated type hints (removed MessageMin where it should be event)
Updated docs, examples and tests

* Updated example, fixed typos

* Removed super init from BuiltinStateDispenser

* Fixed mypy issues after merge, fixed pre/post usage

Updated ABCView to avoid repeated code

* Bumped vkbottle_types to 5.130, Added ABCRawEventView

update event type hints to "Event"

* Added views.abc module to avoid repeated code

Updated type hints

* Fixed User views, vkbottle_types==0.1.24

Renamed some view methods for clarity

dhvcc committed:
Moved tests to module (#350)

Updated pytest/pre-commit configs to reflect new changes

Mypy now ignores tests
Poetry excludes tests from build

dhvcc committed:
Refactor 3.1 (#351)

* Renamed dev/production_tools to dev/production

* Split naming for user/bot views

* Returned import to user.py, dynamic ignores for autoflake

timoniq committed:
Remove old tracked directories *_tools

dhvcc committed:
fixed `BotMessageView.get_state_key` (#352)

mazzz3r committed:
fix blueprints load (#353)

dhvcc committed:
Fix isort skips, same as black's force-exclude (#360)

timoniq committed:
readable exception for deprecated longpoll versions

Commit:
Replace existing `ExceptionFactory` with `CodeException` (#376)

Переработана система хэндлинга исключений. Удалена `CodeErrorFactory`, вместо неё теперь используется `CodeException`, имеющее generic-подобный синтаксис.

Возможности:
 - хэндлинг исключений с любым кодом: `CodeException`;
 - хэндлинг исключения с определённым кодом: `CodeException[1]`;
 - указание нескольких кодов: `CodeException[1, 2, 3]`.

BREAKING CHANGE: новая система создания исключений не поддерживает старый способ передачи кода и описания ошибки в метод `__call__`, вместо этого сначала создаётся новый класс исключения путём передачи кода ошибки в квадратных скобках и инициализируется сообщением ошибки: `VKAPIError(1, "msg")` -> `VKAPIError[1]("msg")`; `SingleError` убрана за ненадобностью.

timoniq committed:
transferring repo

timoniq committed:
update docs

timoniq committed:
refactor middlewares

axd1x8a committed:
Some cosmetic rule changes

axd1x8a committed:
Add `state_dispenser` param for Bot and User init

klalafuda committed:
Update video.py (#383)

KeyError bug fix

axd1x8a committed:
Fixed a bug with overriding raw_event handlers for the same events. (#382)

dhvcc commented on Makefile:
```diff
@@ -43,3 +67,4 @@ publish: ## Publish to PyPi using PYPI_TOKEN
 	@# "; true" is used to ignore command exit code so that rm -rf can execute anyway
 	poetry publish; true
 	rm -rf dist/
+	extract_ignores=$(shell echo "$$(grep ':F401' .flake8 | sed -n 's/\s*\(\S*\):F401/\1,/p' | tr '\n' '\0')")
```
`extract_ignores` сверху уже определен, а это старая некрасивая версия

Commit:
style: format code with `black`

Commit:
add specific `VKAPIError`s such as `CaptchaError`, etc (#380)

Добавлена возможность хэндлинга специфичных ошибок vk с дополнительными полями, таких как `CaptchaError`

BREAKING CHANGE: из `VKAPIError` удалён параметр `raw_error`, так как нет необходимости хранить респонс целиком (дополнительные параметры явно указываются в конструкторе конкретного исключения)

Commit:
refactor(exceptions): change `ErrorHandler` for better code readability, rename decorators

`ErrorHandler` избавлен от сложной логики. `wraps_error_handler` переименован в `catch`. Теперь
соответствующие хендлеры будут вызываться для обработки не только ошибок указанного класса, но и их
подтипов (для удовлетворения LSP). Также добавлена возможность регистрации хендлеров сразу для
нескольких типов ошибок.

Commit:
test: fix `MockedClient` and `test_bot_polling.callback` function

(1) Раньше `MockedClient` неправильно передавал параметры запроса в `callback`, а
`test_bot_polling.callback` неправильно эти параметры принимал. Этот баг не был виден, т.к.
`ErrorHandler` игнорировал ошибки, для которых не зарегистрирован хендлер, поэтому `assert`ы в тесте
также игнорировались.
(2) В `Message.answer` выражение `locals().pop("kwargs")` на самом деле не
удаляло `kwargs` из локальных переменных и `kwargs` попадало в `data`. Этот коммит исправляет этот
баг.

Commit:
docs: fix `ErrorHandler` example

timoniq committed:
Merge pull request #387 from vanburgerberg/refactor-errorhandler

Refactor `ErrorHandler` & fix `test_bot_polling` test

yilbegan committed:
Add `event_data` generator, close #374 (#389)

* Add event_data constructor (#374)

* Add example for event_data constructor (#374)

* Add tests for event_data constructor (#374)

* Fix name shadowing

* Rename `ShowSnackbar` -> `ShowSnackbarEvent`
* Rename `OpenLink` -> `OpenLinkEvent`
* Rename `OpenApp` -> `OpenAppEvent`

* Use `pydantic.BaseModel`

* Reorder imports

* Remove useless tests

Commit:
move logic from `CodeExceptionMeta` to `CodeException` (#388)

Нет необходимости создавать метакласс для решения данной задачи. Вот и не будем.
Ранее при наследовании `CodeException` в `__exceptions__` дочернего класса копировались
конкретизированные исключения родительского класса, что приводило к возможности возврата исключения
неправильного типа из `__class_getitem__`. Теперь при наследовании `CodeException` `__exceptions__`
у дочернего класса будет очищаться.

timoniq committed:
fix failure event crash

INFERNO256 committed:
Added `message.reply` and `message.forward` (#393)

* Add `reply` and `forward` methods to `MessageMin`
* Fix forwading messages to other chats

Co-authored-by: FeeeeK

timoniq committed:
fix blueprint load

timoniq committed:
fix raw event view

kforbro committed:
fix ignore case example (#394)

KurimuzonAkuma committed:
Fix typo (#396)

axd1x8a committed:
Merge branch 'master' into merge

axd1x8a committed:
Fix urls in docs

axd1x8a committed:
Remove old filter converter

axd1x8a committed:
Add NotFilter

axd1x8a committed:
Fix typing

dependabot[bot] committed:
Update mypy requirement from ^0.770 to ^0.910 (#377)

Updates the requirements on [mypy](https://github.com/python/mypy) to permit the latest version.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.770...v0.910)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

axd1x8a committed:
Update typing and fix mypy errors (#399)

Commit:
fixed spelling errors (#397)

axd1x8a committed:
Merge branch 'master' into dev

axd1x8a committed:
Docs for `event_data` generators (#391)

axd1x8a committed:
Update index.md for tools

timoniq committed:
readme update

monosans committed:
Fix typo (evant -> event) (#401)

axd1x8a committed:
Add message splitter

axd1x8a committed:
Fix mypy issue

axd1x8a committed:
Update MessageMin

axd1x8a committed:
Some fixes

axd1x8a committed:
Сontext transfer based on handler signature (#385)

Commit:
fix(exceptions): print `VKAPIError`'s description in traceback (#402)

Теперь описания ошибок `VKAPIError` будут отображаться в трейсбеке

Commit:
fix(exceptions): add `request_params` parameter to `VKAPIError` (#404)

* fix(exceptions): add `request_params` parameter to `VKAPIError`

* docs(exceptions): fix `VKAPIError` handling example

Commit:
refactor(auth): new authorization tool for obtaining user token using login and password (#407)

* refactor(auth): new authorization tool for obtaining user token using login and password

* docs(auth): add documentation for `UserAuth`

axd1x8a committed:
Add user direct auth (#403)

dependabot[bot] committed:
Update pytest-cov requirement from ^2.8 to ^3.0 (#405)

Updates the requirements on [pytest-cov](https://github.com/pytest-dev/pytest-cov) to permit the latest version.
- [Release notes](https://github.com/pytest-dev/pytest-cov/releases)
- [Changelog](https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest-cov/compare/v2.8.0...v3.0.0)

---
updated-dependencies:
- dependency-name: pytest-cov
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

axd1x8a committed:
Fix for undefined user_lp events (#408)

dependabot[bot] committed:
Update flake8 requirement from ^3.7.9 to ^4.0.1 (#409)

Updates the requirements on [flake8](https://github.com/pycqa/flake8) to permit the latest version.
- [Release notes](https://github.com/pycqa/flake8/releases)
- [Commits](https://github.com/pycqa/flake8/compare/3.7.9...4.0.1)

---
updated-dependencies:
- dependency-name: flake8
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Update types-aiofiles requirement from ^0.1.9 to ^0.7.0 (#410)

Updates the requirements on [types-aiofiles](https://github.com/python/typeshed) to permit the latest version.
- [Release notes](https://github.com/python/typeshed/releases)
- [Commits](https://github.com/python/typeshed/commits)

---
updated-dependencies:
- dependency-name: types-aiofiles
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

axd1x8a committed:
Fix blueprints load on windows

axd1x8a committed:
Fix bp_importer test

Commit:
refactor: change `vkbottle.http` for easier use and better readability (#413)

* refactor: change `vkbottle.http` for easier use and better readability

* Fixed cringe refactoring

* Fixed tests

* Fixed requests in auth

* Fixed ClientSession warnings

* Fixed session close in `SingleAiohttpClient`

* Fixed unclosed session warning

Co-authored-by: FeeeeK <26704473+FeeeeK@users.noreply.github.com>

axd1x8a committed:
Fix session close

axd1x8a committed:
Some fixes

axd1x8a committed:
Fix message view event processing without handlers

mironovmeow committed:
Update __init__ in BotLabeler (#416)

* Update __init__ in BotLabeler

* Some fixes

Co-authored-by: FeeeeK <26704473+FeeeeK@users.noreply.github.com>

axd1x8a committed:
Fix comment

axd1x8a committed:
Add vscode to recommended IDE's

axd1x8a committed:
Fix request error with wrong mimetype

axd1x8a committed:
Another fix for wrong mimetype

axd1x8a committed:
Fix for new vkbottle-types

axd1x8a committed:
Fix wrong user raw_event dataclass

dependabot[bot] committed:
Update pytest-asyncio requirement from ^0.15.0 to ^0.16.0 (#421)

Updates the requirements on [pytest-asyncio](https://github.com/pytest-dev/pytest-asyncio) to permit the latest version.
- [Release notes](https://github.com/pytest-dev/pytest-asyncio/releases)
- [Commits](https://github.com/pytest-dev/pytest-asyncio/compare/v0.15.0...v0.16.0)

---
updated-dependencies:
- dependency-name: pytest-asyncio
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

axd1x8a committed:
Fix wrong user raw_event dataclass (again)

axd1x8a committed:
Added `mode` for user polling

axd1x8a committed:
Generalize `ConsistentTokenGenerator` checkup

axd1x8a committed:
Fix typing in `ConsistentTokenGenerator`

axd1x8a committed:
Fix `VBMLRule` init without config

timoniq committed:
Merge branch 'master' into dev

yilbegan committed:
Use string to store state, fix state collisions (#424)

* Use string state in rules (#423)

* Tests for #423

* Move `StatePeer` and `BaseStateGroup` from vkbottle_types (#423)

* Fix documentation for #423

axd1x8a committed:
Some fixes in rules

axd1x8a committed:
Bump version
